torch/export/unflatten.py

```diff
@@ -202,22 +203,53 @@ def __init__(
                 attr_kind=_AttrKind.CONSTANT,
             )
 
-        inputs_to_state: Dict[str, str] = {
-            **self.graph_signature.inputs_to_parameters,
-            **self.graph_signature.inputs_to_buffers,
-            **self.graph_signature.inputs_to_lifted_tensor_constants,
-            **self.graph_signature.inputs_to_lifted_custom_objs,
-        }
+        # This is to handle parameters/buffers that point to the same tensor
+        # object id -> list of (node_name, target_name)
+        consts_map: Dict[int, List[Tuple[str, str]]] = {}
+
+        def add_to_consts_map(obj_id, node_name, target_name):
+            name_list = consts_map.get(obj_id, [])
+            name_list.append((node_name, target_name))
+            consts_map[obj_id] = name_list
```

Comment:
nit: you can use defaultdict to avoid redudndant assigning `name_list` back to `const_map`, when `obj_id` already in the map

```
from collections import defaultldict
const_map: Dict[int, List[Tuple[str, str]]] = defaultdict(list)

def add_to_consts_map(obj_id, node_name, target_name):
    name_list = consts_map.get[obj_id]
    name_list.append((node_name, target_name))
```